### PR TITLE
#1401. [Patterns] Map pattern tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Review by eernstg — COMMENTED** [review]
A few adjustments suggested.

**Comment by sgrekhov** [comment]
All of the requested changes are done. Please rereview

File: LanguageFeatures/Patterns/map_A04_t01.dart [hunk 1]
```diff
@@ -52,8 +59,16 @@ String test1(Map map) {
 //              ^
 // [analyzer] unspecified
 // [cfe] unspecified
-    case {c1: var a, c2: final b} => "";
-//                   ^^
+    case {c1: var a1, c2: final b1} => "";
+//                    ^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    case {3.14: var a2, 3.14: final b2} => "";
```
Comment (eernstg) [inline LanguageFeatures/Patterns/map_A04_t01.dart:66]: Yes! With the new rule about errors for identical keys, `double` keys _are_ covered. Nice! ;-)

File: LanguageFeatures/Patterns/map_A04_t02.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+/// methods
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "../../Utils/expect.dart";
+
+String test1(Map map) {
+  return switch (map) {
+    case {3.14: var a, 3.14: final b} => "a=$a, b=$b";
```
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Patterns/map_A04_t02.dart]: Duplicate `double`s are an error now!

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**